app/controllers/departments_controller.rb

```diff
@@ -50,24 +51,7 @@ def set_department
     @department = Department.find(params[:id])
   end
 
-  def phone_attrs
-    [:id, :body, :label, :_destroy, :type, :contacts_id]
-  end
-
-  def email_attrs
-    [:id, :body, :label, :_destroy, :type, :contacts_id]
-  end
-
-  def contact_attrs
-    [
-      :id, :name, :_destroy, :contactable_id, :contactable_type, :street,
-      :extended, :city, :postal_code,
-      contact_emails_attributes: email_attrs,
-      contact_phones_attributes: phone_attrs
-    ]
-  end
-
   def department_params
-    params.require(:department).permit(user_ids: [], contact_attributes: contact_attrs)
+    params.require(:department).permit(policy(@department).permitted_attributes)
```

Comment:
@nuritnt where where you're difficulties? 
`policy(@department).permitted_attributes` returns the array with allowed parameters. The method is [here](https://github.com/panter/aoz-003/blob/d4924f04aff1927fd0b272ce0bd5726041e7ba51/app/policies/department_policy.rb#L23)

@kitsane how would you evaluate user's permission to set attributes, if not like this?